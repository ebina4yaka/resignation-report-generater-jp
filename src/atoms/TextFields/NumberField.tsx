import { ChangeEvent, ReactElement } from 'react'
import TextField from '@material-ui/core/TextField'
import InputAdornment from '@material-ui/core/InputAdornment'
import useStyles from './useStyles'
import removeNonNumberFromString from '../../libs/removeNonNumberFromString'
import convertNumberFromZenkaku from '../../libs/convertNumberFromZenkaku'
import { TextFieldProps } from './TextFieldProps'

export default function NumberField(
  props: TextFieldProps<number>
): ReactElement {
  const { value, setValue, id, label, endAdornment, isRequired } = props
  const classes = useStyles()

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { target } = event
    const valueNumber = Number(
      removeNonNumberFromString(convertNumberFromZenkaku(target.value))
    )
    setValue(valueNumber)
    target.value = String(valueNumber)
  }

  return (
    <TextField
      required={isRequired}
      id={id}
      label={label}
      value={value}
      className={classes.textField}
      InputProps={{
        endAdornment: (
          <InputAdornment position="end">{endAdornment}</InputAdornment>
        ),
      }}
      InputLabelProps={{
        shrink: true,
      }}
      autoComplete="off"
      onChange={handleChange}
    />
  )
}
