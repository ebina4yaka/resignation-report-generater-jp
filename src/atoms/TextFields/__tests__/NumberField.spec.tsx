import renderer from 'react-test-renderer'
import { act, renderHook } from '@testing-library/react-hooks'
import { fireEvent, queryByAttribute, render } from '@testing-library/react'
import { useState } from 'react'
import NumberField from '../NumberField'

describe('NumberField Tests', () => {
  test('Snapshot test', () => {
    const tree = renderer
      .create(
        <NumberField
          id="testNumber"
          isRequired
          endAdornment="日"
          value={0}
          label="テスト"
          setValue={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Input onChange Handle Test', () => {
    const { result } = renderHook(() => {
      const [value, setValue] = useState(0)
      return { value, setValue }
    })
    const id = 'testNumber'
    const dom = render(
      <NumberField
        id={id}
        isRequired
        endAdornment="日"
        value={result.current.value}
        label="テスト"
        setValue={result.current.setValue}
      />
    )
    const getById = queryByAttribute.bind(null, 'id')

    const inputElement = getById(dom.container, id) as HTMLInputElement

    act(() => {
      fireEvent.change(inputElement, { target: { value: '12' } })
    })

    expect(result.current.value).toBe(12)
  })
})
