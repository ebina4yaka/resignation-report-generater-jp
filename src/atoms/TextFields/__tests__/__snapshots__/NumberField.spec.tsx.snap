// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberField Tests Snapshot test 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-textField-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
    data-shrink={true}
    htmlFor="testNumber"
    id="testNumber-label"
  >
    テスト
    <span
      aria-hidden={true}
      className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
    >
       
      *
    </span>
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoComplete="off"
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
      disabled={false}
      id="testNumber"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="text"
      value={0}
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        日
      </p>
    </div>
  </div>
</div>
`;
